laws/src/main/scala/cats/laws/UnorderedFoldableLaws.scala

```diff
@@ -21,24 +21,6 @@ trait UnorderedFoldableLaws[F[_]] {
       (F.isEmpty(fa) || F.exists(fa)(p))
     } else true // can't test much in this case
 
-  def existsLazy[A](fa: F[A]): Boolean = {
```

Comment:
Foldable extends UnorderedFoldable. To me, the bug here is that we weren’t calling these in UnorderedFoldable tests and that FoldableLaws were not calling through to them as we should do any time we extend a typeclass.

That seems like the better fix to me than moving the law (since we should still want these laws on UnorderedFoldable).

Reply:
It does look that way but actually the default implementations of `exist` and `forall` in `UnorderedFoldable` do not obey these laws. These laws can only be supported in `Foldable` through `foldRight`, hence they should be in `Foldable`

Reply:
We can keep the laws in `UnorderedFoldableTests` if we implement the `CommutativeMonoid[Eval[Boolean]]`s directly in `UnorderedFoldable`.  See https://github.com/rossabaker/cats/commit/4af246d0f4e2377e9e94defe3eedc3ab64cd9dbf.